tests/test_SklearnPipelineWithinPipeline.py

```diff
@@ -0,0 +1,93 @@
+                    steps=[('PCA', PCA(copy=True, iterated_power='auto',
+                            n_components=0.15842105263157896, random_state=None,
+                            svd_solver='auto', tol=0.0, whiten=False)), 
+                           ('Pipeline', Pipeline(memory=None,
+                                steps=[('MinMax scaler', MinMaxScaler(copy=True, feature_range=(0, 3.7209871159509307)))]))])
+
+        data = numpy.array([[0, 0], [0, 0], [1, 1], [1, 1]], dtype=numpy.float32)
+        y = [0, 0, 1, 1]
+        model.fit(data, y)
+        model_onnx = convert_sklearn(model, 'pipelinewithinpipeline', [('input', FloatTensorType([1, 2]))])
+        self.assertTrue(model_onnx is not None)
+        dump_data_and_model(data, model, model_onnx, basename="SklearnPipelinePcaPipelineMinMax")
+        
+    def test_pipeline_pca_pipeline_none_lin(self):
+        
+        model = Pipeline(memory=None,
+                    steps=[('PCA', PCA(copy=True, iterated_power='auto',
+                            n_components=0.15842105263157896, random_state=None,
+                            svd_solver='auto', tol=0.0, whiten=False)), 
+                           ('Pipeline', Pipeline(memory=None,
+                                steps=[('MinMax scaler', MinMaxScaler(copy=True, feature_range=(0, 3.7209871159509307))),
+                                       ('logreg', LogisticRegression())]))])
+
+        data = numpy.array([[0, 0], [0, 0], [1, 1], [1, 1]], dtype=numpy.float32)
+        y = [0, 0, 1, 1]
+        model.fit(data, y)
+        model_onnx = convert_sklearn(model, 'pipelinewithinpipeline', [('input', FloatTensorType([1, 2]))])
+        self.assertTrue(model_onnx is not None)
+        dump_data_and_model(data, model, model_onnx, basename="SklearnPipelinePcaPipelineMinMaxLogReg")
+        
+    @unittest.skip(reason="Type 'tensor(float)' of input parameter (input) of operator (ConstantOfShape) "
+                          "in node (ConstantOfShape) is invalid.")
+    def test_pipeline_pca_pipeline_multinomial(self):
+        
+        model = Pipeline(memory=None,
+                    steps=[('PCA', PCA(copy=True, iterated_power='auto',
+                            n_components=2, random_state=None,
+                            svd_solver='auto', tol=0.0, whiten=False)), 
+                           ('Pipeline', Pipeline(memory=None,
+                                steps=[('MinMax scaler', MinMaxScaler(copy=True, feature_range=(0, 3.7209871159509307))),
+                                       ('MultinomialNB', MultinomialNB(alpha=0.7368421052631579,
+                                                                      class_prior=None, fit_prior=True))]))])
+
+        data = numpy.array([[0, 0, 0], [0, 0, 0.1], [1, 1, 1.1], [1, 1.1, 1]], dtype=numpy.float32)
+        y = [0, 0, 1, 1]
+        model.fit(data, y)
+        model_onnx = convert_sklearn(model, 'pipelinewithinpipeline', [('input', FloatTensorType([1, 2]))])
```

Comment:
The shape of your data on which you fit the model is (4, 3), but you have input set as FloatTensorType([1, 2]) when doing model conversion. I would suggest you do FloatTensorType(data.shape) to avoid such mistakes.

Reply:
The first dimension refers to the number of observations which may vary at prediction. It is usually 1 to call ONNX graph for every observation.